### PR TITLE
Fix ledger controller

### PR DESCRIPTION
**Pull request by Xela101**
### New Pull Request Submissions:

1. [x] Have you followed the guidelines in our [Contributing document](https://github.com/atlascity/Community/tree/master/CONTRIBUTING.md)?
2. [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
3. [x] I have added tests to cover my changes.
4. [x] All new and existing tests passed.
5. [x] Have you lint your code locally prior to submission?
6. [x] Does your code follows the code style of this project?
7. [x] Does your change require a change to the documentation.
    - [ ] I have updated the documentation accordingly.
9. [x] Have you added an explanation of what your changes do and why you'd like us to include them?
10. [x] Have you inserted a keyword and link to the issues the PR closes in its descriptions (ex `closes #1`) ?
11. [x] Is you branch up to date, have you integrated all the latest changes from develop and resolved conflicts ?

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4482731&bid=15125329)

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4482731&bid=15125329)

File: src/Catalyst.Core.Lib/Util/CidHelper.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static class CidHelper
 
         public static Cid CreateCid(MultiHash multiHash)
         {
-            return new Cid {Version = 1, Hash = multiHash, ContentType = "dag-pb", Encoding = Encoding};
```
Comment (nshCore) [inline src/Catalyst.Core.Lib/Util/CidHelper.cs]: Is their reasoning behind this change?

File: src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs [hunk 2]
```diff
@@ -39,37 +42,46 @@ public sealed class LedgerController : Controller
...
         public async Task<JsonResult> GetLatestDelta(DateTime? asOf)
         {
             var latest = _deltaHashProvider.GetLatestDeltaHash(asOf?.ToUniversalTime());
-            byte[] dfsContent = null;
             try
             {
                 using (var fullContentStream = await _dfs.ReadAsync(latest))
                 {
-                    dfsContent = await fullContentStream.ReadAllBytesAsync(CancellationToken.None);
+                    var contentBytes = await fullContentStream.ReadAllBytesAsync(CancellationToken.None);
```
Comment (monsieurleberre) [inline src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs]: conifgureawait = false?
I feel like I am Codacy 😂

File: src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs [hunk 3]
```diff
@@ -39,37 +42,46 @@ public sealed class LedgerController : Controller
...
+                        Success = true,
+                        DeltaHash = latest,
+                        Delta = delta
+                    });
                 }
             }
             catch (Exception e)
             {
-                _logger.Error(e, "Failed to find dfs content for delta as of {asOf} at {dfsTarget}", asOf, latest);
+                var errorMessage = $"Failed to find dfs content for delta as of {asOf} at {latest}";
```
Comment (monsieurleberre) [inline src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs]: you lost the structured logging by passing a raw string here... no big deal, but worth keeping in mind (cf. https://serilog.net/)

File: src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs (hunk 2 above)
Comment (nshCore) [reply src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs]: yep all async operations need to .configureawait(false)

File: src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs (hunk 3 above)
Comment (Xela101) [reply src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs]: Cheers, I thought it was just a simple format wrapper, didn't realise it provided a better format structure. Updated based on your feedback.

File: src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs (hunk 2 above)
Comment (Xela101) [reply src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs]: done

**Comment by nshCore** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Catalyst.Framework/Catalyst.Framework/pullRequest?prid=4482731&bid=15125329)

File: src/Catalyst.Core.Lib/Util/CidHelper.cs (hunk 1 above)
Comment (Xela101) [reply src/Catalyst.Core.Lib/Util/CidHelper.cs]: image

I saw dfs/ipfs was returning Content-Type raw not dag-pb.

**Pull request merged**